profiling/proto/service/service.proto

```diff
@@ -0,0 +1,91 @@
+// Copyright 2019 gRPC authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+// This package defines the proto messages and RPC services exposed by gRPC for
+// profiling management. A reference client implementation to interact with
+// this service is provided as a command-line application. This service can be
+// used to toggle profiling on and off and retrieve stats from a gRPC
+// application.
+option go_package = "google.golang.org/grpc/profiling/proto/service";
+
+// EnableRequest defines the fields in a /Profiling/Enable method request to
+// toggle profiling on and off within a gRPC program.
+message EnableRequest {
+  // Setting this to true will enable profiling. Setting this to false will
+  // disable profiling.
+  bool enabled = 1;
+}
+
+// EnableResponse defines the fields in a /Profiling/Enable method response.
+message EnableResponse {
+}
+
+// GetStreamStats defines the fields in a /Profiling/GetStreamStats method
+// request to retrieve stream-level stats in a gRPC client/server.
+message GetStreamStatsRequest {
+}
+
+// A TimerProto measures the start and end of execution of a component within
+// gRPC that's being profiled. It includes a tag and some additional metadata
+// to identify itself.
+message TimerProto {
+  // timer_tag is an arbitrary string that is used to tag a timer for
+  // identification purposes later.
+  string timer_tag = 1;
+
+  // begin_sec and begin_nsec are the start epoch second and nanosecond,
+  // respectively, of the component profiled by this timer in UTC. begin_nsec
+  // must be a non-negative integer.
+  int64 begin_sec = 2;
```

Comment:
Not sure if this is necessarily a good idea, but...

https://github.com/protocolbuffers/protobuf/blob/master/src/google/protobuf/timestamp.proto

If it helps, there's a library that deals with these:

https://godoc.org/github.com/golang/protobuf/ptypes

Reply:
@easwars raised this too (https://github.com/grpc/grpc-go/pull/3160#discussion_r344923697) FWIW, and I considered using timestamp.proto previously, but I favoured explicitly defining the sec and nsec because it was more straightforward and self-contained (no external dependencies).

There's absolutely no difference between the two other than aesthetics (which is subjective, of course), and I'd be perfectly fine with moving to timestamp.proto if insisted upon, but changing it at this point involves a few changes in profiling/cmd (even though the on-wire representation will be unchanged).

I'll admit that my case for sticking to `begin_sec` over timestamp.proto is essentially a weak argument of "I think this looks cleaner and I'm a bit lazy to change this everywhere" lol.